# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dietlibc/elf-stt-reg.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

On Fri, 18 Jul 2003 09:24:03 +0200 (CEST), Rene Rebe <rene.rebe@gmx.net> said:

> Hi, On: Thu, 17 Jul 2003 18:58:17 +0100,
> alan.rock@glaramara.freeserve.co.uk wrote:

>> In function `calculate_deps32': moduleops_core.c:118:
>> `STT_REGISTER' undeclared (first use in this function)
>> 
>> This is because the version of elf.h supplied by dietlibc doesn't
>> define STT_REGISTER
>> 

> No - it compiled in the past - and it must compile with dietlibc
> since we use it in the first stage of the installation for module
> loading.

OK - try this patch against dietlibc then

--- ./include/elf.new.h	2002-02-21 20:45:21.000000000 +0000
+++ ./include/elf.h	2003-07-18 10:54:48.000000000 +0100
@@ -142,6 +142,8 @@
 #define STT_SECTION	3
 #define STT_FILE	4
 
+#define STT_REGISTER	13		/* Global register reserved to app. */
+
 #define ELF32_ST_BIND(x) ((x) >> 4)
 #define ELF32_ST_TYPE(x) (((unsigned int) x) & 0xf)
 


